File: 2017 COMLAN/RangeValueMonitor/src-weaved/c/foobar.c
#include "lib/lib.h"
#include "range_values_monitor1.h"

double bar() {
   
   return 1.0;
}


double foo() {
   double a = 0;
   for(int i = 0; i < 1000; i++) {
      /* monitoring a */
      a += bar();
      monitor1_range_update(0, a);
   }
   /* monitoring a */
   a = lib_call(a);
   monitor1_range_update(0, a);
   
   return a;
}


int main() {
   monitor1_range_init();
   foo();
   monitor1_print_ranges();
}

File: 2017 COMLAN/RangeValueMonitor/src-weaved/matlab_with_api/foobar.m
function foobar()
   monitor1_range_init();
   foo();
   monitor1_range_print();
end

function [a] = foo()
   % monitoring a
   a = 0;
   monitor1_range_update(1, a);
   for i = 0:5
      % monitoring a
      a = a + bar();
      monitor1_range_update(1, a);
   end
   % monitoring a
   a = libcall(a);
   monitor1_range_update(1, a);
end

function [result] = bar()
   % monitoring result
   result = 1.0;
   monitor1_range_update(2, result);
end

